src/components/App.tsx

```diff
@@ -751,6 +752,35 @@ function App() {
     if (messages) navigator.clipboard.writeText(messages);
   };
 
+  /*//////////////////////////////////////////////////////////////
+                         RENAME NODE LOGIC
+  //////////////////////////////////////////////////////////////*/
+
+  // TODO: Prevent node name to be overridden when GPT node is edited
+
+  const [nodeToRename, setNodeToRename] = useState<Node>();
+
+  const toggleShowRenameInput = () => {
+    const selectedNode = nodes.find((node) => node.selected);
```

Comment:
should fallback to using `selectedNodeId` using `??` here, as a user can have no nodes selected on the canvas but still be editing a specific node